Cache strokeStyle and fillStyle for performance

wxs opened the pull request:
Fixes #2011 

Many code paths repeatedly reset or access the stroke or the fill, for
instance when drawing shapes every call to vertex() queries the fill and
stroke styles. This seems to be quite an expensive operation, and for
almost every use-case is not changing between calls to vertex.

Here I replace all direct accesses to the canvas' drawingContext's
strokeStyle and fillStyle to go through accessors that cache the result.
This shows significant performance improvements, at least on Chrome on
Mac.

lmccart commented on src/core/p5.Renderer2D.js:
```diff
@@ -953,13 +946,29 @@ p5.Renderer2D.prototype.strokeWeight = function(w) {
 
 p5.Renderer2D.prototype._getFill = function(){
-  return this.drawingContext.fillStyle;
+  return this.cachedFillStyle;
```
can we call this this._cachedFillStyle? similar with this._cachedStrokeStyle?

lmccart commented:
this is great! see the one comment i left inline.

wxs commented:
Hey @lmccart I've addressed your comment!

I've also made a couple more changes.
 - The fillStyle was being set directly in `p5.Font.js`, I've fixed that
 - I found a further performance improvement, due to the fact that setting a fillStyle or a strokeStyle doesn't necessarily return the same string back if you access it later. This slightly complicates the tests.
 -  Some tests fail intermittently for me due to a timeout. This makes development really frustrating so I increased the timeout time

lmccart commented:
this is a great improvement, thank you!

kadamwhite commented:
This is excellent!
